zilliqa/src/api/eth.rs

```diff
@@ -1035,8 +1035,95 @@ fn blob_base_fee(_params: Params, _node: &Arc<Mutex<Node>>) -> Result<()> {
 
 /// eth_feeHistory
 /// Returns the collection of historical gas information
-fn fee_history(_params: Params, _node: &Arc<Mutex<Node>>) -> Result<()> {
-    Err(anyhow!("API method eth_feeHistory is not implemented yet"))
+fn fee_history(params: Params, node: &Arc<Mutex<Node>>) -> Result<FeeHistory> {
+    let mut params = params.sequence();
+    let block_count: String = params.next()?;
+    let block_count = if let Some(block_count) = block_count.strip_prefix("0x") {
+        u64::from_str_radix(block_count, 16)?
+    } else {
+        block_count.parse::<u64>()?
+    };
+
+    let block_count = block_count.min(1024);
+
+    if block_count == 0 {
+        return Ok(FeeHistory::default());
+    }
+
+    let newest_block: BlockNumberOrTag = params.next()?;
+    let reward_percentiles: Option<Vec<f64>> = params.optional_next()?.unwrap_or_default();
```

Comment:
Did you mean to add `unwrap_or_default` here? I believe this means the API is now expecting a `Option<Option<Vec<f64>>>` which doesn't make much sense.